src/libs/defaultRedirection/defaultRedirection.ts

```diff
@@ -0,0 +1,129 @@
+import AsyncStorage from '@react-native-async-storage/async-storage'
+
+import CozyClient, { Q, deconstructCozyWebLinkWithSlug } from 'cozy-client'
+
+import { formatRedirectLink } from '/libs/functions/formatRedirectLink'
+import { logger } from '/libs/functions/logger'
+import { getErrorMessage } from '/libs/functions/getErrorMessage'
+import strings from '/constants/strings.json'
+
+export const log = logger('AppBundle')
+
+export const NAVIGATION_APP_SLUG = 'home'
+
+export const DEFAULT_REDIRECTION_DELAY_IN_MS = 3000
+
+export interface InstanceSettings {
+  data: {
+    attributes: {
+      default_redirection: string
+    }
+  }
+}
+
+export const isDefaultRedirectionUrlNavigationApp = (
+  defaultRedirectionUrl: string,
+  client: CozyClient
+): boolean => {
+  const subdomainType = client.capabilities.flat_subdomains ? 'flat' : 'nested'
+
+  const { slug } = deconstructCozyWebLinkWithSlug(
+    defaultRedirectionUrl,
+    subdomainType
```

Comment:
Note for another PR: we re-write this `flat_subdomains ? 'flat' : 'nested'` thing everywhere, we definitively needs a helper to convert `flat_domain:boolean` into `subdomainType:'flat'|'nested'`

Reply:
I TOTALLY agree. I noted it for myself yesterday. I am more thinking on cozy-client side because it also happens on front app. Something like client.getSubdomainType => return 'flat' or 'nested' (no hook no async). @Crash-- what do you think ?